src/lib.rs

```diff
@@ -3296,13 +3344,18 @@ impl Connection {
                     Err(e) => return Err(e),
                 };
 
-                stream.recv.push(data)?;
+                let rx_inc = stream.recv.push(data)?;
 
                 if stream.is_readable() {
                     self.streams.mark_readable(stream_id, true);
                 }
 
-                self.rx_data += data_len;
+                // Check for flow control limits.
+                if self.rx_data + rx_inc > self.max_rx_data {
+                    return Err(Error::FlowControl);
+                }
+
+                self.rx_data = self.rx_data.saturating_add(rx_inc);
```

Comment:
`stream::push()` describes itself with "This also takes care of enforcing stream flow control limits", so is there a reason not to do the check inside push? It seems like it would be easier

Reply:
This part is checking "connection level FC limit". stream::push() checks "stream level FC limit" as described.

Reply:
Makes sense. I think the comment was confusing me so maybe better to just say "check the connection flow control limit" or something

Reply:
Updated the comment.